standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -3254,31 +3247,38 @@ protected List<Partition> getJdoResult(GetHelper<List<Partition>> ctx) throws Me
   public Partition getPartitionWithAuth(String catName, String dbName, String tblName,
       List<String> partVals, String user_name, List<String> group_names)
       throws NoSuchObjectException, MetaException, InvalidObjectException {
-    boolean success = false;
-    try {
-      openTransaction();
-      MPartition mpart = getMPartition(catName, dbName, tblName, partVals, null);
-      if (mpart == null) {
-        commitTransaction();
-        throw new NoSuchObjectException("partition values="
-            + partVals.toString());
+    return new GetHelper<Partition>(catName, dbName, tblName, false, true) {
```

Comment:
why refactor?

Reply:
It may throw exception, so we need such logic:
```java
try {
  openTransaction();

  success = commitTransaction();
} finally {
  if (!success) rollbackTransaction();
}
```

Reply:
does it even run in txn? where do we open it?